### PR TITLE
Fixes for bitcoin-cash and general usability

### PR DESCRIPTION
**Pull request by gandrewstone**
Fix unknown argument error in bitcoin-cli, 
change testnet op_return testnet sunset to new time, 
fix getheaders sync problem.
add a hidden configurator that defeats the getblocktemplate RPC's safety checks.  These checks disallow you from mining blocks if you chain tip is too old or if you are not connected.  However, these checks can make it impossible to resume mining a test network if sufficient time has passed since you last mined it.

### CODE REVIEW

**Comment by sickpig** [comment]
check failed due to formatting error in `unlimited.h`, see 
https://travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/jobs/256502961#L1294

@gandrewstone please run clang on it.

File: src/main.cpp [hunk 1]
```diff
@@ -7421,9 +7422,10 @@ bool SendMessages(CNode *pto)
         if (!state.fSyncStarted && !pto->fClient && !fImporting && !fReindex)
         {
             // Only actively request headers from a single peer, unless we're close to today.
-            if ((nSyncStarted == 0 && fFetch) || pindexBestHeader->GetBlockTime() > GetAdjustedTime() - 24 * 60 * 60)
+            if ((nSyncStarted == 0 && fFetch) ||
+                chainActive.Tip()->GetBlockTime() > GetAdjustedTime() - SINGLE_PEER_REQUEST_MODE_AGE)
             {
-                const CBlockIndex *pindexStart = pindexBestHeader;
+                const CBlockIndex *pindexStart = chainActive.Tip(); // pindexBestHeader;
```
Comment (ptschip) [inline src/main.cpp:7441]: i think you can take the comment out?  // pindexBestHeader

File: src/main.cpp [hunk 2]
```diff
@@ -3838,10 +3837,15 @@ CBlockIndex *AddToBlockIndex(const CBlockHeader &block)
         pindexNew->pprev = (*miPrev).second;
         pindexNew->nHeight = pindexNew->pprev->nHeight + 1;
         pindexNew->BuildSkip();
+        // BU If the prior block or an ancestor has failed, mark this one failed
+        if (pindexNew->pprev && pindexNew->pprev->nStatus & BLOCK_FAILED_MASK)
+            pindexNew->nStatus |= BLOCK_FAILED_CHILD;
```
Comment (gandrewstone) [inline src/main.cpp:3848]: This code is needed to ensure that an invalid header chain is properly marked as invalid as new headers are added to it.

Comment (ptschip) [reply src/main.cpp:3848]: I just meant the  comment on line 7428  "// pindexBestHeader"  , that's the old code you commented out...i think you don't need to leave it there...it's just a nit

**Comment by ptschip** [comment]
i tested with this, syncd to the correct chain on the testnet and also tried out  the validatechainhistory feature...it works

**Comment by sickpig** [comment]
this one fail on `wallet.py`
https://travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/jobs/256626160#L2047

File: src/main.cpp (hunk 1 above)
Comment (gandrewstone) [reply src/main.cpp:7441]: done [EDIT] actually, I'll commit it in a different PR so this one can be merged...

**Comment by gandrewstone** [comment]
no wallet code was changed so this is likely not related to this PR

**Pull request merged**